swaylock/main.c

```diff
@@ -510,6 +511,11 @@ int main(int argc, char **argv) {
 		create_layer_surface(surface);
 	}
 
+	if (state.args.daemonize) {
+		wl_display_roundtrip(state.display);
```

Comment:
Could this possibly have the same problem as verified, e.g. we'd need to wait till all surfaces are not dirty anymore?
(well, I guess forking is faster than password validation so it's probably fine; but we'd ideally want to make sure the compositor has created the keyboard-exclusive layer surface before forking; not just that we asked the compositor to create it)

Reply:
The roundtrip ensures that the compositor has received the request.

We could probably do more stuff like checking the compositor hasn't immediately closed the layer surface (I'm not even sure what we do in this case?) and checking we have sent buffers. The latter will be a more involved change.

Reply:
Well if this change is about "waiting for the lockscreen to be "safe" before forking" then it's still lying a bit, that's what I meant yes.
Meh I guess this can wait.